src/model/chat.test.ts

```diff
@@ -91,7 +91,10 @@ describe('ChatModel', () => {
   });
 
   it('implements extend', async () => {
-    const chatModel = new ChatModel({
+    const chatModel = new ChatModel<{
```

Comment:
just a DX remark; this "feels" weird to me. sorry that's not concrete

Reply:
yeah it does feel weird and probably wouldn't ever be specified inline like that. i'll update the test to better reflect how i expect it to be used given that it's the only "documentation" right now.

Reply:
@transitive-bullshit is this better? https://github.com/dexaai/dexter/pull/34/commits/858045e74082f8f6c37ee4cc18be4e4a41105f27

Reply:
lgtm 🚀